python/ray/serve/handle.py

```diff
@@ -1,27 +1,29 @@
 import asyncio
 import concurrent.futures
-import threading
-from typing import Any, Coroutine, Dict, Optional, Union
+from dataclasses import dataclass, field
+from typing import Any, Coroutine, Dict, Optional, Type, TypeVar, Union
 
 import ray
 from ray.serve.context import TaskContext
 from ray.serve.router import RequestMetadata, Router
 from ray.serve.utils import get_random_letters
 from ray.serve.exceptions import RayServeException
 
-global_async_loop = None
 
+@dataclass
+class HandleOptions:
+    method_name: str = "__call__"
+    shard_key: Optional[str] = None
+    http_method: str = "GET"
+    http_headers: Dict[str, str] = field(default_factory=dict)
 
-def create_or_get_async_loop_in_thread():
-    global global_async_loop
-    if global_async_loop is None:
-        global_async_loop = asyncio.new_event_loop()
-        thread = threading.Thread(
-            daemon=True,
-            target=global_async_loop.run_forever,
-        )
-        thread.start()
-    return global_async_loop
+
+# Use a global singleton
+class Default(object):
+    pass
+
+
+DEFAULT_VALUE = Default()
```

Comment:
I think this should be an immutable object to avoid the strange "modify the default args" issue -> can we just use a string instead?

Reply:
We can't use string nor None because they are all acceptable values. Because we own the DEFAULT_VALUE and not returning it to anywhere, we don't have the "modify the default args" issue.

Reply:
but i did changed it to enum to make it immutable though!